File: EduStack-Backend/src/entities/Course.ts
import { CourseLevel } from '@prisma/client';

export interface Course {
  id: string;
  title: string;
  description: string;
  thumbnail: string;
  price: number;
  category: string;
  level: CourseLevel;
  duration: number; // in weeks
  instructorId: string;
  isPublished: boolean;
  createdAt: Date;
  updatedAt: Date;
}

export interface CreateCourseData {
  title: string;
  description: string;
  thumbnail: string;
  price: number;
  category: string;
  level: CourseLevel;
  duration: number;
  instructorId: string;
}

export interface UpdateCourseData {
  title?: string;
  description?: string;
  thumbnail?: string;
  price?: number;
  category?: string;
  level?: CourseLevel;
  duration?: number;
  isPublished?: boolean;
}

export interface CourseWithInstructor extends Course {
  instructor: {
    id: string;
    fullName: string;
    avatar?: string;
  };
}

export interface CourseWithStats extends CourseWithInstructor {
  enrollmentCount: number;
  averageRating: number;
  reviewCount: number;
}

File: EduStack-Backend/EduStack.Core/Entities/Course.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EduStack.Core.Entities;

public class Course
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(255)]
    public string Title { get; set; } = string.Empty;

    [Required]
    public string Description { get; set; } = string.Empty;

    [Required]
    [MaxLength(500)]
    public string Thumbnail { get; set; } = string.Empty;

    [Required]
    [Column(TypeName = "decimal(10,2)")]
    public decimal Price { get; set; }

    [Required]
    [MaxLength(100)]
    public string Category { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    public string Level { get; set; } = "BEGINNER"; // BEGINNER, INTERMEDIATE, ADVANCED

    [Required]
    public int Duration { get; set; } // in weeks

    [Required]
    public Guid InstructorId { get; set; }

    public bool IsPublished { get; set; } = false;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation Properties
    [ForeignKey("InstructorId")]
    public virtual User Instructor { get; set; } = null!;

    public virtual ICollection<Enrollment> Enrollments { get; set; } = new List<Enrollment>();
    public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
    public virtual ICollection<Progress> Progress { get; set; } = new List<Progress>();
    public virtual ICollection<Lesson> Lessons { get; set; } = new List<Lesson>();
}
